src/meta/processors/index/CreateEdgeIndexProcessor.cpp

```diff
@@ -96,7 +97,15 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
     }
 
     if (checkIndexExist(fields, item)) {
-      resp_.code_ref() = nebula::cpp2::ErrorCode::E_EXISTED;
+      if (ifNotExists) {
+        resp_.code_ref() = nebula::cpp2::ErrorCode::SUCCEEDED;
+        cpp2::ID thriftID;
```

Comment:
maybe use tools like  `resp_.id_ref() = to(edgeIndex, EntryType::INDEX);` ?

Reply:
Just extract the id from the index item.